checkout/interfaces/controller/apicontroller.go

```diff
@@ -0,0 +1,244 @@
+package controller
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+
+	"flamingo.me/flamingo-commerce/v3/cart/domain/cart"
+	placeorderDomain "flamingo.me/flamingo-commerce/v3/cart/domain/placeorder"
+	"flamingo.me/flamingo-commerce/v3/checkout/application"
+	"flamingo.me/flamingo-commerce/v3/checkout/domain/placeorder/process"
+
+	"flamingo.me/flamingo/v3/framework/flamingo"
+	"flamingo.me/flamingo/v3/framework/web"
+
+	cartApplication "flamingo.me/flamingo-commerce/v3/cart/application"
+	"flamingo.me/flamingo-commerce/v3/cart/domain/decorator"
+	"flamingo.me/flamingo-commerce/v3/checkout/application/placeorder"
+)
+
+type (
+	// APIController for checkout rest api
+	APIController struct {
+		responder            *web.Responder
+		placeorderHandler    *placeorder.Handler
+		cartService          *cartApplication.CartService
+		logger               flamingo.Logger
+		decoratedCartFactory *decorator.DecoratedCartFactory
+	}
+
+	// startPlaceOrderResult result of start place order
+	startPlaceOrderResult struct {
+		UUID string
+	}
+
+	// placeOrderContext infos
+	placeOrderContext struct {
+		Cart         *cart.Cart
+		OrderInfos   *placedOrderInfos
+		State        string
+		StateData    interface{}
```

Comment:
maybe using type StateData here as well, is also type interface{} but would match the core place order context